httpc.go

```diff
@@ -461,10 +461,10 @@ func (r *Request) RunWithContext(ctx context.Context) error {
 		// In both cases simply continue and have the client handle the (potentially
 		// cancelled) context (that way the same error is returned no matter _when_
 		// the context was cancelled)
-		ticker := time.NewTicker(r.retryIntervals[i])
+		timer := time.NewTimer(r.retryIntervals[i])
```

Comment:
add `defer timer.Stop()` so the resources get collected when `context.Done()`  is read before the timer finishes.

Reply:
Fair point - although `timer.Stop()` only needs to be called if it hasn't triggered yet, so I'll add it to the `Done()` path of the context:
```
		case <-ctx.Done():
			timer.Stop()
```